lib/Service/BoardService.php

```diff
@@ -683,6 +648,87 @@ private function enrichBoards(array $boards, bool $fullDetails = true): array {
 		return $boards;
 	}
 
+	private function cloneCards(Board $board, Board $newBoard, bool $withAssignments = false, bool $withLabels = false, bool $withDueDate = false, bool $moveCardsToLeftStack = false, bool $restoreArchivedCards = false): void {
+		// TODO: Undelete cards
+		// TODO: Copy attachments (or not?)
+		// TODO: Copy comments (or not?)
+		// TODO: Move to specific column
+
+		$stacks = $this->stackMapper->findAll($board->getId());
+		usort($stacks, function (Stack $a, Stack $b) {
+			return $a->getOrder() - $b->getOrder();
+		});
+
+		$newStacks = $this->stackMapper->findAll($newBoard->getId());
+		usort($stacks, function (Stack $a, Stack $b) {
+			return $a->getOrder() - $b->getOrder();
+		});
+
+		$i = 0;
+		foreach ($stacks as $stack) {
+			$cards = $this->cardMapper->findAll($stack->getId());
+			$archivedCards = $this->cardMapper->findAllArchived($stack->getId());
+
+			/** @var Card[] $cards */
+			$cards = array_merge($cards, $archivedCards);
+
+			foreach ($cards as $card) {
+				$targetStackId = $moveCardsToLeftStack ? $newStacks[0]->getId() : $newStacks[$i]->getId();
+
+				// Create a cloned card.
+				$newCard = new Card();
```

Comment:
TODO: check if we have a better mechansim for cloning to just set the new related ids?

Reply:
I've checked that and it seems to be not possible. As, when inserting, only fields marked as updated get Insterted into the DB, and according to [this](https://github.com/nextcloud/server/blob/500a9cb7244d3072f83eebc878e27833572e707d/lib/public/AppFramework/Db/QBMapper.php#L122) there is no other way to set all fields as updated other than using the corresponding setter.